aws_google_auth/google.py

```diff
@@ -331,6 +284,53 @@ def parse_saml(self):
 
         return base64.b64decode(saml_element)
 
+    def handle_captcha(self, sess, payload):
+        response_page = BeautifulSoup(sess.text, 'html.parser')
+
+        # Collect ProfileInformation, SessionState, signIn, and Password Challenge URL
+        profile_information = response_page.find('input', {
+            'name': 'ProfileInformation'
+        }).get('value')
+        session_state = response_page.find('input', {
+            'name': 'SessionState'
+        }).get('value')
+        sign_in = response_page.find('input', {'name': 'signIn'}).get('value')
+        passwd_challenge_url = response_page.find('form', {
+            'id': 'gaia_loginform'
+        }).get('action')
+
+        # Update the payload
+        payload['SessionState'] = session_state
+        payload['ProfileInformation'] = profile_information
+        payload['signIn'] = sign_in
+        payload['Passwd'] = self.config.password
+
+        # Get all captcha challenge tokens and urls
+        captcha_container = response_page.find('div', {'class': 'captcha-container'})
+        captcha_logintoken = captcha_container.find('input', {'name': 'logintoken'}).get('value')
+        captcha_url = captcha_container.find('input', {'name': 'url'}).get('value')
+        captcha_logintoken_audio = captcha_container.find('input', {'name': 'logintoken_audio'}).get('value')
+        captcha_url_audio = captcha_container.find('input', {'name': 'url_audio'}).get('value')
+
+        # Open captcha image
+        with requests.get(captcha_url) as url:
```

Comment:
@mide @louahola 
https://github.com/cevoaustralia/aws-google-auth/pull/91#discussion_r207663245

Changed urlopen to requests here